app/scripts/modules/core/application/modal/pagerDutySelectField.directive.html

```diff
@@ -0,0 +1,14 @@
+<div class="form-group row">
+  <div class="col-sm-3 sm-label-right">PagerDuty</div>
+  <div class="col-sm-9">
+
+    <ui-select ng-model="vm.component[vm.field]" class="form-control input-sm"  required>
+      <ui-select-match placeholder="Select a PagerDuty Service">{{$select.selected.name}}</ui-select-match>
+      <ui-select-choices repeat="pagerDuty.integration_key as pagerDuty in vm.pagerDutyServices | filter: $select.search">
```

Comment:
When editing an existing application, I'm guessing the `pagerDutyServices` won't be available immediately (since it's coming in via an ajax call), so I _think_ the ui-select component will remove the existing value.

You can get around this by putting an `ng-if="$ctrl.servicesLoaded"` attribute on the `<ui-select` tag, then set that field in the controller after adding the `pagerDutyServices` to `this`.